runbot_travis2docker/models/runbot_build.py

```diff
@@ -355,3 +359,53 @@ def _local_cleanup(self, cr, uid, ids, context=None):
             if build.docker_container:
                 build.docker_rm_container()
                 build.docker_rm_image()
+
+    def schedule(self, cr, uid, ids, context=None):
+        res = super(RunbotBuild, self).schedule(cr, uid, ids, context=context)
+        for build in self.browse(cr, uid, ids, context=context):
+            if (build.state == 'running' and build.job == 'job_30_run' and
+                    not build.docker_executed_commands):
+                session = requests.Session()
+                if build.repo_id.token:
+                    session.auth = (build.repo_id.token, 'x-oauth-basic')
+                    session.headers.update({'PRIVATE-TOKEN':
+                                            build.repo_id.token})
+                try:
+                    executed = True
+                    match_object = re.search('([^/]+)/([^/]+)/([^/.]+(.git)?)',
+                                             build.repo_id.base)
+                    url = 'https://api.github.com/repos/%s/%s/commits/%s' % (
+                        match_object.group(2), match_object.group(3),
+                        build.name)
+                    response = session.get(url)
+                    response.raise_for_status()
+                    json = response.json()
+                    cmd = ["docker", "exec", "--user=root",
+                           build.docker_container, "/etc/init.d/ssh", "start"]
+                    _logger.info('Start ssh server : ' + ' '.join(cmd))
+                    output = subprocess.check_output(cmd)
+                    if 'done.' not in output:
+                        _logger.exception('Fail to start ssh server: %s',
+                                          output)
+                    else:
+                        _logger.info(output)
+                    cmd = ["docker", "exec", "--user=odoo",
+                           build.docker_container, "curl",
+                           "https://github.com/%s.keys" %
+                           json['author']['login'], "-o",
+                           "/home/odoo/.ssh/authorized_keys"]
+                    _logger.info('Copy keys of github : ' + ' '.join(cmd))
+                    subprocess.call(cmd)
+                    cmd = ["docker", "exec", "--user=odoo",
```

Comment:
This validation should be moved to tests section
IMHO here we are running `docker-exec` commands unnecessarily